examples/oauth_flow.rs

```
@@ -0,0 +1,286 @@
+    println!("{auth_url}");
+    println!();
+
+    println!("🚀 Starting local HTTP server on 127.0.0.1:8080...");
+    println!("📋 Next Steps:");
+    println!("1. The server is now running and waiting for OAuth callback");
+    println!("2. Open the authorization URL above in your browser");
+    println!("3. Grant permissions to your application");
+    println!("4. You'll be redirected back and the flow will complete automatically");
+    println!();
+
+    // Channel to receive the authorization code
+    let (tx, rx) = oneshot::channel();
+    let tx = Arc::new(tokio::sync::Mutex::new(Some(tx)));
+
+    // Create the HTTP service
+    let make_svc = make_service_fn(move |_conn| {
+        let tx = tx.clone();
+        async move {
+            Ok::<_, Infallible>(service_fn(move |req| {
+                let tx = tx.clone();
+                handle_request(req, tx)
+            }))
+        }
+    });
+
+    // Start the server
+    let addr = ([127, 0, 0, 1], 8080).into();
+    let server = Server::bind(&addr).serve(make_svc);
+
+    // Start the server and wait for the OAuth callback
+    println!("⏳ Waiting for OAuth callback...");
+    println!("💡 If your browser doesn't open automatically, copy and paste the URL above");
+    println!();
+
+    // Try to open the URL in the default browser
+    if let Err(e) = webbrowser::open(&auth_url) {
+        println!("⚠️  Could not open browser automatically: {e}");
+        println!("📋 Please manually open the URL above in your browser");
+    }
+
+    // Wait for the callback using select to handle both server and receiver
+    let auth_code = tokio::select! {
+        result = rx => {
+            match result {
+                Ok(code) => {
+                    println!("🔄 OAuth callback received, processing...");
+                    code
+                }
+                Err(_) => {
+                    println!("❌ Failed to receive authorization code");
+                    return Ok(());
```

should probably return an Err here, so the exit status is non-zero